src/Hosting.Services.Remoting/RemotingListenerBuilder.cs

```diff
@@ -20,23 +21,21 @@ public abstract class RemotingListenerBuilder<TService> : IListenerBuilder<TServ
 		public string Name { get; }
 
 		// done internal for unit tests
-		internal FabricTransportRemotingListenerSettings? Settings { get; }
+		internal FabricTransportRemotingListenerSettings Settings { get; }
 
 		/// <summary>
 		/// Constructor
 		/// </summary>
+		/// <remarks>Transport security settings should be defined in TransportSettings of a service's settings.xml before creating a listener</remarks>
 		protected RemotingListenerBuilder(
 			string name,
 			FabricTransportRemotingListenerSettings? settings = null)
 		{
-			if (settings == null)
+			if (settings == null && !FabricTransportRemotingListenerSettings.TryLoadFrom("TransportSettings", out settings))
```

Comment:
FabricTransportRemotingListenerSettings.LoadFrom could be used here either, but I decided that throwing a very explicit exception was more desirable.